armi/reactor/blueprints/__init__.py

```diff
@@ -321,18 +324,80 @@ def _prepConstruction(self, cs):
                 self.assemblies[aDesign.name] = a
             if currentCount != 0:
                 assemblies.setAssemNumCounter(currentCount)
-
-            self._checkAssemblyAreaConsistency(cs)
-
             runLog.header("=========== Verifying Assembly Configurations ===========")
+            self._checkAssemblyAreaConsistency(cs)
 
-            # pylint: disable=no-member
-            getPluginManagerOrFail().hook.afterConstructionOfAssemblies(
-                assemblies=self.assemblies.values(), cs=cs
+            # if assemblies are defined in blueprints, handle meshing
+            # assume finest mesh is reference
+            assemsByNumBlocks = sorted(
+                self.assemblies.values(),
+                key=lambda a: len(a),
+                reverse=True,
             )
+            referenceAssembly = assemsByNumBlocks[0] if assemsByNumBlocks else None
+
+            if not cs["detailedAxialExpansion"]:
+                # make the snap lists so assems know how to expand
+                nonUniformAssems = [
+                    Flags.fromStringIgnoreErrors(t) for t in cs["nonUniformAssemFlags"]
+                ]
+                # prepare core for mesh snapping during axial expansion
+                for a in self.assemblies.values():
+                    if any(a.hasFlags(f) for f in nonUniformAssems):
+                        continue
+                    a.makeAxialSnapList(referenceAssembly)
+            if not cs["inputHeightsConsideredHot"]:
+                # expand axial heights from cold to hot
+                self._coldDimsToHot(referenceAssembly, cs["detailedAxialExpansion"])
```

Comment:
@john-science happy to move this stuff else where and have a 1 line call to hit here.